start_admin_sshd.sh

```diff
@@ -21,8 +21,10 @@ get_user_data_keys() {
     # Extract the keys from user-data json
     local raw_keys
     local key_type="${1:?}"
-    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]' "${user_data}" 2>/dev/null); then
-      log "Failed to parse ${key_type} from ${user_data}"
+    # ? signifies an optional object identifier-index and doesn't return a 'failed to iterate' error
+    # in the event that a key_type is missing. If jq returns nothing, then that is caught by the -z.
+    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}") \
+    || [[ -z "${raw_keys}" ]]; then
```

Comment:
nit: I'd prefer a less complex conditional check; assigning a variable and then checking it in a subsequent test is a bit too clever for my taste.

```suggestion
    raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}")
    if [ "${?}" -gt 0 ] || [ -z "${raw_keys}" ] ; then
```

Reply:
That doesn't work with `set -e`; the shell would exit after the assignment before it can even get to the `$?` check.